pkg/bff/activity_test.go

```diff
@@ -20,6 +22,103 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+func (s *bffTestSuite) TestNetworkActivity() {
+	require := s.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// If there is no activity in the database, should return zeros
+	rep, err := s.client.NetworkActivity(ctx)
+	require.NoError(err, "could not get network activity")
+	require.Len(rep.MainNet, 30, "should have 30 days of activity for mainnet")
+	require.Len(rep.TestNet, 30, "should have 30 days of activity for testnet")
+	assertNoCounts(require, rep)
+
+	// Add an empty month to the database but with no data
+	now := time.Now()
+	month := &models.ActivityMonth{
+		Date: now.Format(models.MonthLayout),
+	}
+	require.NoError(s.DB().UpdateActivityMonth(ctx, month), "could not create activity month")
+	rep, err = s.client.NetworkActivity(ctx)
+	require.NoError(err, "could not get network activity")
+	assertNoCounts(require, rep)
+
+	// Add some activity to the database
+	month.Add(&activity.NetworkActivity{
+		Network: activity.MainNet,
+		Activity: activity.ActivityCount{
+			activity.LookupActivity: 1,
+		},
+		Timestamp: now,
+		Window:    time.Minute * 5,
+	})
+	require.NoError(s.DB().UpdateActivityMonth(ctx, month), "could not update activity month")
+	rep, err = s.client.NetworkActivity(ctx)
+	require.NoError(err, "could not get network activity")
+	require.Len(rep.MainNet, 30, "should have 30 days of activity for mainnet")
+	require.Len(rep.TestNet, 30, "should have 30 days of activity for testnet")
+	require.Equal(uint64(1), rep.MainNet[29].Events, "should have one mainnet event")
+	require.Zero(rep.TestNet[28].Events, "should have zero testnet events")
+
+	// Create some more activity
+	acv := &activity.NetworkActivity{
+		Network: activity.TestNet,
+		Activity: activity.ActivityCount{
+			activity.LookupActivity: 3,
+		},
+		Timestamp: now.AddDate(0, 0, -29),
+		Window:    time.Minute * 5,
+	}
+
+	// Create a new month if less than 29 days in this month
```

Comment:
Probably need to think of a better way to test this that's not day dependent.

Reply:
The best way to do that is to create a top level function in in the package:

```go
var currentTime = time.Now 

func MockTime(f func() time.Time) {
    currentTime = f
}
```

There are monkey patching libraries like https://github.com/undefinedlabs/go-mpatch -- but if the only concern is patching `time.Now()` then the above is the easiest method.